pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -158,11 +163,45 @@ func isRequiredConfigPresent(config []byte, isNotSingleNode bool) error {
 			if !ok {
 				return fmt.Errorf("%v is not a slice", strings.Join(requiredPath, "."))
 			}
-			if len(configValSlice) < 3 {
-				return fmt.Errorf("%v has less than three endpoints: %v", strings.Join(requiredPath, "."), configValSlice)
+
+			if err := etcdEndpointsPresent(c.configMapLister, configValSlice, requiredPath); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// etcdEndpointsPresent compares the list of endpoints exposed by
+// cluster-etcd-operator with the list of URLs in the etcd-servers config. If
+// less than two etcd live endpoints (not localhost and bootstrap) are present
+// in the config, it returns an error because the configuration does not
+// statisfy HA requirements.
+func etcdEndpointsPresent(configMapLister corev1listers.ConfigMapLister, config []interface{}, configPath []string) error {
+	etcdEndpointsCM, err := configMapLister.ConfigMaps(etcdEndpointNamespace).Get(etcdEndpointName)
+	if err != nil {
+		return err
+	}
+
+	var liveEndpoints []string
+	for _, val := range config {
+		url, ok := val.(string)
+		if !ok {
+			return fmt.Errorf("%v is not a string slice", strings.Join(configPath, "."))
+		}
+
+		for _, etcdEndpoint := range etcdEndpointsCM.Data {
+			if strings.Contains(url, etcdEndpoint) {
```

Comment:
Based on https://github.com/openshift/cluster-etcd-operator/blob/8ffe9611d1afd27c7af4273d6e7f28cef6f07952/pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go#L131 `etcd-bootstrap` member(s) are excluded from the list of endpoints.